### ISSUE TITLE
libxml2 not included: No module named libxml2mod

### ISSUE THREAD

**Issue by bsavelev** [pyinstaller/pyinstaller#1555]
Hello!
I have a test app
```python
import libxml2
print (libxml2.XML_SCHEMAP_INTERNAL)
```
run pyinstaller (from git  3.0.dev2-231-gf411960) 
```../pyinstaller.py -y --clean -D --hidden-import=libxml2mod ./t.py```

try to run and gets and error
```python
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/build/pyinstaller/PyInstaller/loader/pyimod03_importers.py", line 363, in load_module
    exec(bytecode, module.__dict__)
  File "/usr/lib/python2.7/dist-packages/libxml2.py", line 1, in <module>
    import libxml2mod
ImportError: No module named libxml2mod
t returned -1
```

**Comment by htgoebel** [comment]
`libxml2mod` is imported by `libxml2` in the very first line. So there should be no need for a `--hiddenimport` (neither in he command line nor in a hook). Something else is wrong here.

**Comment by bsavelev** [comment]
Yes, I add hidden-import for sure that libxml2mod not included in result... So I cant add this module force too. Its strange)

**Comment by htgoebel** [comment]
I tracked this down: `libxm2mod.so` is found as expected, but removed from the list of binaries in `PyiModuleGraph._make_toc()` (see [line 308](https://github.com/pyinstaller/pyinstaller/blob/6ffbb5180557/PyInstaller/depend/analysis.py#L308)). I'm assigning this to @matysek as he implemented this part in 812f5f2bd99c9

**Comment by matysek** [comment]
@htgoebel That makes an assumption that python module name won't: start with `lib` and having extension `.so`. But `libxml` is the corner case :-)

**Comment by bsavelev** [comment]
It have prefix ```lib``` because builded with libtool, I think. And python binding to libxml2 is not the only one

**Comment by matysek** [comment]
@bsavelev That's what I found out.

**Comment by matysek** [comment]
@bsavelev Try this version, if that fixes it: 825ff6868cf949e99ef08f2b3230c5b08a268bb0

**Comment by bsavelev** [comment]
3.0.dev6-25-gd6350ee

It works!
